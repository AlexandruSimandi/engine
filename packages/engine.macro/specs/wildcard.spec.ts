import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "babel-plugin-macros";

const macroFile = "'@c11/engine.macro'";
const config = require("./babelConfig.json");

pluginTester({
  plugin,
  babelOptions: { filename: __filename, ...config },
  formatResult: (result: any) => {
    return prettier.format(result, {
      parser: "babel",
    });
  },
  tests: {
    "should support Wildcard": {
      code: `
        import { Wildcard, view } from ${macroFile};
        const foo: view = ({
          id = Wildcard,
          boo = Observe.foo[Arg.id].foo,
          bar = Observe.foo[Wildcard].bar
        }) => {}
      `,
      snapshot: true,
    },
  },
});
