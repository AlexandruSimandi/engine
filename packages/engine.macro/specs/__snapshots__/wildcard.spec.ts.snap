// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros should support Wildcard: should support Wildcard 1`] = `

import { Wildcard, view } from '@c11/engine.macro';
const foo: view = ({
  id = Wildcard,
  boo = Observe.foo[Arg.id].foo,
  bar = Observe.foo[Wildcard].bar
}) => {}

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _engine = require("@c11/engine.producer");

var _engineView = require("@c11/engine-view");

var foo = (0, _engineView.view)({
  args: {
    type: "STRUCT",
    value: {
      id: {
        type: "VALUE",
        value: {
          type: "CONST",
          value: _engine.Wildcard,
        },
      },
      boo: {
        type: "OBSERVE",
        path: [
          {
            type: "CONST",
            value: "foo",
          },
          {
            type: "INTERNAL",
            path: ["id"],
          },
          {
            type: "CONST",
            value: "foo",
          },
        ],
      },
      bar: {
        type: "OBSERVE",
        path: [
          {
            type: "CONST",
            value: "foo",
          },
          {
            type: "CONST",
            value: _engine.Wildcard,
          },
          {
            type: "CONST",
            value: "bar",
          },
        ],
      },
    },
    meta: {},
  },
  fn: function fn(_ref) {
    var id = _ref.id,
      boo = _ref.boo,
      bar = _ref.bar;
  },
});


`;
